src/tools/pub/dependencies.md

```diff
@@ -442,6 +439,32 @@ To identify dependencies in your app or package that
 aren't on the latest stable versions,
 use [`dart pub outdated`][].
 
+### Use tight dev_dependencies
+
+Dependencies can be loose, specifying dependency at the highest version digit
+like `^2.0.0`, or tight, specifying dependency at the lowest version digit like
+`^1.2.4`.
+
+In your `pubspec.yaml` file, follow these recommendations.
+
+* Set regular `dependencies` loose to increase the chance of a version solver success.
+* Set [`dev_dependencies`][dev-dep] tighter as they lack transitive concerns.
```

Comment:
If we can drop the note on normal dependencies, I think we should just combine this with the above text to better colocate what a tight dependency constraint is and why you should use them for dev dependencies.

Reply:
Also if we keep using the wording "transitive concerns", could we add an example or two of those concerns? I'm afraid the terminology will be less familiar and potentially unclear about what concerns are being referred to.

I'd be happy to help expand here if you need any ideas. Let me know!

Reply:
It's at the top of the page.